### PR TITLE
Log incompatible configuration properties

### PR DESCRIPTION
**Pull request by stIncMale**
Specifically, when the CDC handler is configured
(https://docs.mongodb.com/kafka-connector/current/sink-connector/configuration-properties/cdc/)
configuration properties for post processors
https://docs.mongodb.com/kafka-connector/current/sink-connector/configuration-properties/post-processors/
and configuration properties for the ID strategy
are ignored.

KAFKA-277

### CODE REVIEW

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java [hunk 1]
```diff
@@ -597,7 +596,6 @@ public NamespaceMapper getNamespaceMapper() {
     String prefix = format("%s%s.", TOPIC_OVERRIDE_PREFIX, topic);
     List<String> topicOverrides =
         props.keySet().stream().filter(k -> k.startsWith(prefix)).collect(Collectors.toList());
-    logObsoleteProperties(topicOverrides);
```
Comment (stIncMale) [inline src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java:600]: I realized that `MongoSinkTopicConfig.validateAll` is called from `ConfigDef.validateAll` (see `MongoSinkConfig.createConfigDef`) with the same `props` as those passed to `ConfigDef.validateAll`. I.e., this call results in duplicate logging.

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java [hunk 2]
```diff
@@ -18,7 +18,31 @@
...
 package com.mongodb.kafka.connect.sink;
 
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.CHANGE_DATA_CAPTURE_HANDLER_CONFIG;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.CHANGE_DATA_CAPTURE_HANDLER_DEFAULT;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DELETE_ON_NULL_VALUES_CONFIG;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DELETE_ON_NULL_VALUES_DEFAULT;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DOCUMENT_ID_STRATEGY_CONFIG;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DOCUMENT_ID_STRATEGY_OVERWRITE_EXISTING_CONFIG;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DOCUMENT_ID_STRATEGY_OVERWRITE_EXISTING_DEFAULT;
+import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.DOCUMENT_ID_STRATEGY_UUID_FORMAT_CONFIG;
```
Comment (rozza) [inline src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java:28]: I think the topic configuration should be responsible for logging all topic based configuration.  The top level sink configuration should delegate to each topic configuration.

**Review by rozza — CHANGES_REQUESTED** [review]

File: src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java (hunk 2 above)
Comment (rozza) [reply src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java:28]: The benefits are the topic based configuration manages its own validation and not the higher level sink config.  It seems to be recoupling the design by pushing this upwards.

Comment (stIncMale) [reply src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java:28]: > the topic based configuration manages its own validation

I don't think this is the case: the validation is currently done (even before this PR) by the static code. I.e., there is no object that represents topic-specific configuration that has an instance method validating it. Instead, there is a static method `MongoSinkTopicConfig.validateAll` doing this, and this static method is called from `ConfigDef.validateAll` defined in `MongoSinkConfig`.

In other words, the proposal will result only in restructuring the static code. No responsibility will be moved from from an object to a different object, because validation is done by static code, not instance methods. I.e, it is simply impossible to say that the topic based configuration will manage its own validation, because the object representing that configuration is created after the static code does the validation.

Comment (stIncMale) [reply src/main/java/com/mongodb/kafka/connect/sink/MongoSinkConfig.java]: We discussed this over Zoom, and agreed to move the logic into a separate class. It was moved to a new `SinkConfigSoftValidator` class, what exposes methods `logObsoleteProperties` and `logIncompatibleProperties`, used from `ConfigDef.validateAll`.

**Review by rozza — APPROVED** [review]
LGTM
